spec/features/edit_document_associations_spec.rb

```diff
@@ -3,94 +3,69 @@
 require "spec_helper"
 
 RSpec.feature "Edit document associations" do
-  DocumentTypeSchema.all.each do |schema|
-    scenario "User edits associations to a #{schema.name}" do
-      @schema = schema
-
-      given_there_is_a_document_with_associations
-      when_i_visit_the_document_page
-      and_i_click_on_edit_associations
-      then_i_can_see_the_current_selections
-      when_i_edit_the_associations
-      then_i_can_view_the_associations
-      and_the_preview_creation_succeeded
-    end
-
-    def given_there_is_a_document_with_associations
-      @document = create(:document, title: "Title", document_type: @schema.id)
-      @association_schemas = @document.document_type_schema.associations
-
-      @association_schemas.each do |schema|
-        publishing_api_has_linkables(linkables, document_type: schema.document_type)
-      end
-
-      initial_associations = @association_schemas.map do |schema|
-        [schema.id, [linkables[2]["content_id"]]]
-      end
-
-      @document.update(associations: Hash[initial_associations])
-    end
-
-    def when_i_visit_the_document_page
-      visit document_path(@document)
-    end
-
-    def and_i_click_on_edit_associations
-      click_on "Edit associations"
-    end
+  let(:initial_association) { { "content_id" => SecureRandom.uuid, "internal_name" => "Initial association" } }
+  let(:association_to_select_1) { { "content_id" => SecureRandom.uuid, "internal_name" => "Association to select 1" } }
+  let(:association_to_select_2) { { "content_id" => SecureRandom.uuid, "internal_name" => "Association to select 2" } }
+
+  scenario "User edits associations to a document" do
+    given_there_is_a_document_with_associations
```

Comment:
Slight inconsistency with this step and the API down version. I'd be happy with 'given_there_is_a_document'.